### DISCUSSION TITLE
Feature request: Deployments point to wrong commits

### DISCUSSION THREAD

**Discussion by szatanjl** [community/community#57013] [Actions]
### Select Topic Area

Bug

### Body

I have following git history:

commit0 <-- main
commit1
...
commit2 <-- tag v1.0.0
...

I have a workflow to deploy tags into environment.  It is a workflow_dispatch and I run it by hand.  One of the parameters is tag to deploy.  And so I tell the workflow to deploy tag v1.0.0 to dev env.  It build correct tag and deploys correct software.  But at no point did I tell GitHub actions what it is deploying.  Because it is workflow_dispatch the github.sha is set to `main`, not to `v1.0.0`.

And so then on "Deployment history" GitHub is telling me that I just deployed commit0 (main) onto dev env, which is not true.  I deployed v1.0.0.

There should be an option to specify what am I deploying (defaulting to github.sha if not specified).  For example

```
environment:
  name: dev
  commit: ${{ inputs.sha }}
```

**Comment by fmal** [comment]
@szatanjl yeah that would be nice, in the meantime i recommend using https://github.com/chrnorm/deployment-action action which let's you provide sha as `ref` input.

You could also use `gh api` command (https://docs.github.com/en/rest/deployments/deployments?apiVersion=2022-11-28#create-a-deployment) and provide sha via `ref` field:

```sh
  gh api repos/OWNER/REPO/deployments \
    --method POST \
    --header "Accept: application/vnd.github.v3+json" \
    --field ref='SHA' \
    --field environment='dev' 
```
